jwst/background/background_sub.py

```diff
@@ -91,16 +91,24 @@ def get_subset_array(self, other):
         if jmax < jmin:
             jmax = jmin
 
+        # To ensure that we can mix and match subarray obs, take
+        # the x/y shape from self.data. To ensure we can work
+        # with mismatched NINTS, if we have that third dimension
+        # use the value from other
+        data_shape = list(self.data.shape)
```

Comment:
why is casting to a list necessary?

Reply:
This was my suggested change - it's because self.data.shape is a tuple, which can't be modified after creation.  We can either cast it to a list to modify the first member, or create a new tuple from other.data.shape + self.data.shape.